src/hardware/mixer.cpp

```diff
@@ -877,9 +898,11 @@ bool MixerChannel::TryParseAndSetCustomFilter(const std::string &filter_prefs)
 			return false;
 		}
 
-		const auto max_cutoff_freq_hz = (do_zoh_upsampler
-		                                         ? zoh_upsampler.target_freq
-		                                         : sample_rate) / 2 - 1;
+		const auto do_zoh_upsample = (resample_method ==
+		                              ResampleMethod::ZeroOrderHoldAndResample);
```

Comment:
Up top, we've `resample_method` holding one of the three methods and `do_resample` controlling _if_ resampling is performed or not.

This variable name here. `do_zoh_upsample` is a combination of both: the doing and the method, however, the _doing_ is implied. 

Maybe it should be 

```c++
const auto do_zoh_upsample = do_resample && 
    (resample_method == ResampleMethod::ZeroOrderHoldAndResample);
```

Or if the _doing_ is programmatically guaranteed by the time we're in this part of the code, let's check it:

``` c++
assert(do_resample);
```

Reply:
It's kinda assumed, but better to include it in the condition, yes.